### PR TITLE
 Added babel parser support for javascript parsing (updated)

### PR DESCRIPTION
**Pull request by Knogobert**
This was originally a PR from @ziemenz at #72 

Updated to 2.14.0, added some unit tests, updated the readme.
Should be manually tested a bit more first

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "gettext-extract": "./src/extract-cli.js"
   },
   "dependencies": {
-    "@babel/core": "^7.7.5",
+    "@babel/core": "^7.11.5",
```
Comment (vperron) [inline package.json:35]: that could be squashed with the previous commit :)

File: src/javascript-extract.js [hunk 2]
```diff
@@ -108,8 +73,59 @@ function getGettextEntriesFromJavaScript(script, parser = 'auto') {
...
+  case 'babel':
+    extractedEntries = getGettextEntriesFromJavaScript(parseJSGettextWithBabel(script));
+    break;
+
+  default:
+    try {
+      extractedEntries = getGettextEntriesFromJavaScript(parseJSGettextWithAcorn(script));
+    } catch (e) {
+      console.log(`[${PROGRAM_NAME}] will switch extracting using acorn as parser and use babel instead`); // eslint-disable-line no-console
+      extractedEntries = getGettextEntriesFromJavaScript(parseJSGettextWithBabel(script));
```
Comment (vperron) [inline src/javascript-extract.js:125]: seems to me that you could use `getGettextEntriesFromJavaScript` only once at the end of the function, whoich would make it way clearer.

**Review by vperron — APPROVED** [review]
agreed, with a very minor comment :)

I think it's fair to also just squash all those commits together, it's goint to be way more atomic this way. Except maybe for the small style / typo changes that are unrelated.

Thank you so much for your contribution !

File: package.json (hunk 1 above)
Comment (Knogobert) [reply package.json:35]: Done

File: src/javascript-extract.js (hunk 2 above)
Comment (Knogobert) [reply src/javascript-extract.js]: Ah, smart idea! Implemented now

**Comment by Knogobert** [comment]
I have updated according to your comments, good stuff!

**Pull request merged**